parser_library/src/context/hlasm_statement.h

```diff
@@ -53,14 +53,16 @@ struct hlasm_statement
     const semantics::deferred_statement* access_deferred() const;
     semantics::deferred_statement* access_deferred();
 
-    virtual position statement_position() const = 0;
+    position statement_position() const { return stmt_range_ref().start; }
 
+    virtual const range& stmt_range_ref() const = 0;
     virtual std::span<const diagnostic_op> diagnostics() const = 0;
 
-    virtual ~hlasm_statement() = default;
-
 protected:
-    hlasm_statement(const statement_kind kind);
+    explicit hlasm_statement(const statement_kind kind)
+        : kind(kind)
+    {}
+    ~hlasm_statement() = default;
```

Comment:
Should this be `virtual`? 

```suggestion
    virtual ~hlasm_statement() = default;
```

Reply:
All the statements are released via shared_ptr, it was never needed.